huntsman/observatory.py

```diff
@@ -195,6 +200,27 @@ def analyze_recent(self):
 
         return self.current_offset_info
 
+    def require_coarse_focus(self):
+        '''
+        Return True if too much time has elapsed since the previous focus, else False.
+        '''
+        if self.last_focus_time is None:
+            return True
+        if utils.current_time() - self.last_focus_time > self._focus_frequency:
+            return True
+        return False
+
+    def autofocus_cameras(self, *args, **kwargs):
+        '''
+        Override autofocus_cameras to update the last focus time.
+        '''
+        result = super().autofocus_cameras(*args, **kwargs)
+
+        # Update last focus time
+        self.last_focus_time = utils.current_time()
```

Comment:
I'd be slightly stricter and insist on there having been a fine focus recently, not either coarse or fine.

```suggestion
        coarse = kwargs.get("coarse", False)
        if not coarse:
            self.last_focus_time = utils.current_time()
```